File: src/main/java/fr/jonathanwinandy/javaweb/app/Application.java
package fr.jonathanwinandy.javaweb.app;


import com.vaadin.terminal.ThemeResource;
import com.vaadin.ui.*;



public final class Application extends com.vaadin.Application{


    private HorizontalLayout hlayout;
    private Window mainwindow;
    private Label richText;
    private Window detailwindow;
    private static final ThemeResource ICON = new ThemeResource("../runo/icons/16/help.png");
  

    @Override
      public void init() {



        setTheme("mytheme");


        detailwindow = new Window("Inspector");
        VerticalLayout layout = (VerticalLayout) detailwindow.getContent();
        layout.setMargin(true);
        layout.setSpacing(true);
        detailwindow.addComponent(new Label("Content in the inspector window"));

        Button open = new Button("Inspector", new Button.ClickListener() {

            public void buttonClick(Button.ClickEvent event) {
                if (detailwindow.getParent() != null) {

                    getMainWindow().removeWindow(detailwindow);
                } else {

                    getMainWindow().addWindow(detailwindow);
                    detailwindow.setPositionX(30);
                    detailwindow.setPositionY(100);
                    detailwindow.setHeight("40%");

                }
            }
        });
        open.setIcon(ICON);


        VerticalLayout globalLayout= new VerticalLayout();


        final VerticalLayout mainLayout = new VerticalLayout();
        mainLayout.setWidth("750px");
        mainLayout.setSpacing(true);




        globalLayout.addComponent(mainLayout);
        globalLayout.setComponentAlignment(mainLayout, Alignment.MIDDLE_CENTER);

        richText = new Label("<h1>Opups</h1>");
        richText.setContentMode(Label.CONTENT_XHTML);

        mainLayout.addComponent(richText);
        mainLayout.setComponentAlignment(richText, Alignment.MIDDLE_LEFT);


        mainwindow = new Window();
        setMainWindow(mainwindow);
        mainwindow.setBorder(1);
        mainwindow.addComponent(globalLayout);


        HorizontalLayout toolbar = new HorizontalLayout();
        toolbar.addComponent(open);
        toolbar.setSizeFull();
        toolbar.setComponentAlignment(open, Alignment.TOP_LEFT);
        mainLayout.addComponent(toolbar);


    }







}

File: src/main/scala/app/Application.scala
package fr.jonathanwinandy.javaweb.app

import com.vaadin.terminal.ThemeResource
import org.apache.commons.configuration.PropertiesConfiguration
import twitter4j.auth.AccessToken
import twitter4j.TwitterFactory
import com.vaadin.ui._
import collection.JavaConversions

final object Application {
  lazy val ICON: ThemeResource = new ThemeResource("../runo/icons/16/help.png")
}

final class Application extends com.vaadin.Application {
  override def init: Unit = {
    setTheme("mytheme")
    detailwindow = new Window("Inspector")
    val layout = detailwindow.getContent.asInstanceOf[VerticalLayout]
    layout.setMargin(true)
    layout.setSpacing(true)
    detailwindow.addComponent(new Label("Content in the inspector window"))
    val open = new Button("Inspector", new Button.ClickListener {
      def buttonClick(event: Button#ClickEvent): Unit = {
        if (detailwindow.getParent != null) {
          getMainWindow.removeWindow(detailwindow)
        }
        else {
          getMainWindow.addWindow(detailwindow)
          detailwindow.setPositionX(30)
          detailwindow.setPositionY(100)
          detailwindow.setHeight("40%")
        }
      }
    })
    open.setIcon(Application.ICON)

    val richText = new Label("<h1>Opups <i>AHOY</i></h1>") {
      setContentMode(Label.CONTENT_XHTML)
    }

    val toolbar = new HorizontalLayout {
      setSizeFull
      addComponent(open)
      setComponentAlignment(open, Alignment.TOP_LEFT)
    }


    val twitterlayout = new VerticalLayout
    twitter(twitterlayout)

    val mainLayout = new VerticalLayout {
      addComponent(richText)
      addComponent(toolbar)
      addComponent(twitterlayout)
      setComponentAlignment(richText, Alignment.MIDDLE_LEFT)
      setWidth("750px")
      setSpacing(true)
    }




    val globalLayout = new VerticalLayout {
      addComponent(mainLayout)
      setComponentAlignment(mainLayout, Alignment.MIDDLE_CENTER)
    }


    val mainwindow = new Window {
      addComponent(globalLayout)
      setBorder(1)
    }

    this.setMainWindow(mainwindow)


  }

  def twitter(layout: VerticalLayout): Unit = {
    val config = new PropertiesConfiguration("twitter4j.properties")
    val accessToken = new AccessToken(config.getString("twitter.user.token"), config.getString("twitter.user.tokensecret"))
    val factory = new TwitterFactory()
    val twitter = factory.getInstance(accessToken)
    val statuses = twitter.getFriendsTimeline;
    for (status <- JavaConversions.asScalaBuffer(statuses)) {
      layout.addComponent(new Label(status.getUser.getName + ":" +
        status.getText) {
        setHeight("40px")
      })
    }

  }


  private var hlayout: HorizontalLayout = null
  private var detailwindow: Window = null
}